mmv1/third_party/terraform/services/oracledatabase/data_source_oracle_database_autonomous_database.go

```diff
@@ -0,0 +1,36 @@
+package oracledatabase
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+func DataSourceOracleDatabaseAutonomousDatabase() *schema.Resource {
+	dsSchema := tpgresource.DatasourceSchemaFromResourceSchema(ResourceOracleDatabaseAutonomousDatabase().Schema)
+	tpgresource.AddRequiredFieldsToSchema(dsSchema, "location", "autonomous_database_id")
+	tpgresource.AddOptionalFieldsToSchema(dsSchema, "project")
+	return &schema.Resource{
+		Read:   dataSourceOracleDatabaseAutonomousDatabaseRead,
+		Schema: dsSchema,
+	}
+
+}
+
+func dataSourceOracleDatabaseAutonomousDatabaseRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*transport_tpg.Config)
+
+	id, err := tpgresource.ReplaceVars(d, config, "projects/{{project}}/locations/{{location}}/autonomousDatabases/{{autonomous_database_id}}")
+	if err != nil {
+		return fmt.Errorf("Error constructing id: %s", err)
+	}
+	err = resourceOracleDatabaseAutonomousDatabaseRead(d, meta)
+	if err != nil {
+		return err
+	}
+	d.SetId(id)
```

Comment:
After Read, return an error if `id` is null. That indicates we've hit a not found error- Read will return an error if we hit an unexpected 403 or 5XX, but a 404 (or 403 for some resources) will succeed, returning a resource with no id, flagging that Terraform should clear it from state.

This block is standard: https://github.com/GoogleCloudPlatform/magic-modules/blob/72b43deb6fe6a0678ebbc5bc189a6733bdf4e1f2/mmv1/third_party/terraform/services/compute/data_source_google_compute_region_ssl_certificate.go#L44-L46